clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractMembershipManager.java

```diff
@@ -269,6 +271,14 @@ public int memberEpoch() {
         return memberEpoch;
     }
 
+    /**
+     * @return Instance ID used by the member when joining the group. If non-empty, it will indicate that
+     * this is a static member.
+     */
+    public Optional<String> groupInstanceId() {
```

Comment:
This method is used by the sub class (`ConsumerHeartbeatRequestManager`) only, so maybe we don't need to add it to the parent class (`AbstractMembershipManager`)

Reply:
Good catch. It was necessary for it to be in the abstract class at some point in the refactoring process, but it certainly isn't needed now.

Reply:
I agree with the move, and I did think about it the first time but then thought of KStreams (KIP-1071). I expect Streams groups will need this too, and we'll probably have to move it back up right? (just sharing what I had in mind, ok to leave it only where we need it for now)

Reply:
Yup. It's a simple thing to move back again when we need to. I expect streams groups might change things a bit more broadly.